src/App.tsx

```diff
@@ -1,40 +1,48 @@
-import { Routes, Route } from 'react-router-dom'
+import { useState } from 'react'
 
-import { HomePage } from './pages/home-page/HomePage'
-import { BookDetails } from './pages/book-details/BookDetails'
-import { FooterMenu } from './components/footer-menu/FooterMenu'
-import { ProfilePage } from './pages/profile-page/ProfilePage'
-import { SideMenu } from './components/side-menu/SideMenu'
+import { Routes, Route, useLocation } from 'react-router-dom'
+
+import { HomePage } from './pages/homePage/HomePage'
+import { BookDetails } from './pages/bookDetails/BookDetails'
+import { FooterMenu } from './components/footerMenu/FooterMenu'
+import { ProfilePage } from './pages/profilePage/ProfilePage'
+import { SideMenu } from './components/sideMenu/SideMenu'
 import { ConditionalWrapper } from './components/ConditionalWrapper'
-import { HeaderMenu } from './components/header-menu/HeaderMenu'
-import { SearchBar } from './components/search-bar/SearchBar'
+import { HeaderMenu } from './components/headerMenu/HeaderMenu'
+import { SearchBar } from './components/searchBar/SearchBar'
+import { LoginPage } from './pages/loginPage/LoginPage'
 
-import './app.css'
+import './App.css'
 
 function App() {
+  const location = useLocation()
+  const [ authenticated, setAuthenticated ] =  useState<boolean>(false)
   return (
     <div className='app'>
       <header className='app-header'>
-        <ConditionalWrapper
-          condition={screen.availWidth > 768}
-          wrapper={(children) => (
-            <HeaderMenu>
-              <SearchBar />
-            </HeaderMenu>
-          )}
-        >
-          <SearchBar />
-        </ConditionalWrapper>
+        {location.pathname !== '/login' && (
+          <ConditionalWrapper
+            condition={screen.availWidth > 768}
+            wrapper={(children) => (
+              <HeaderMenu isAuthenticated={authenticated} setAuthenticated={setAuthenticated}>
+                <SearchBar />
+              </HeaderMenu>
+            )}
+          >
+            <SearchBar />
+          </ConditionalWrapper>
+        )}
       </header>
       <div className='app-view'>
-        <SideMenu />
+        {location.pathname !== '/login' && <SideMenu isAuthenticated = {authenticated}/>}
         <Routes>
           <Route path='/' element={<HomePage />} />
           <Route path='details' element={<BookDetails />} />
-          <Route path='profile' element={<ProfilePage />} />
+          <Route path='profile' element={<ProfilePage setAuthenticated={setAuthenticated} />} />
+          <Route path='login' element={<LoginPage setAuthenticated={setAuthenticated} />} />
```

Comment:
You didn't make any Protected routes? Can not logged in user access all the pages?